shared/lib/swaps-utils.js

```diff
@@ -139,6 +141,10 @@ const getBaseUrlForNewSwapsApi = (type, chainId) => {
   if (gasApiTypes.includes(type)) {
     return `${gasApiBaseUrl}/networks/${chainIdDecimal}`; // Gas calculations are in its own repo.
   }
+  const tokenApiTypes = ['blockedTokens'];
+  if (tokenApiTypes.includes(type)) {
```

Comment:
Where is this API called?

Reply:
It's called when the Send state is initialized: https://github.com/MetaMask/metamask-extension/blob/develop/ui/ducks/send/send.js#L760

Reply:
Okay thanks. This would be a privacy degradation as it would result in network requests to an external service, for all users of the send flow.

We will have to consider how we can allow users to opt-out or opt-in to the functionality before the network request is made

Reply:
I see. Maybe we can use the "Basic functionality" setting to fetch the data conditionally? 
![Screenshot 2024-06-04 at 2 32 52 PM](https://github.com/MetaMask/metamask-extension/assets/100321200/83e33488-8b56-4b74-9567-be9a771d04fe)

Sent you a calendar invite for a call tomorrow